### DISCUSSION TITLE
SOCKS + UC mode?

### DISCUSSION THREAD

**Discussion by GTWHTX** [seleniumbase/SeleniumBase#2341] [Q&A] [answered]
Is there's any way to connect proxy with UC mode? I've read that wire and uc not compatible with each other, but if im trying to connect proxy like that - Driver(proxy="login:password@socks5://ip:port") it gives an error: ERR_SOCKS_CONNECTION_FAILED

**Comment by mdmintz** [comment, marked as answer]
Duplicate of https://github.com/seleniumbase/SeleniumBase/discussions/2116#discussioncomment-7027095

**Reply by GTWHTX** [reply]
in this case i just get ERR_EMPTY_RESPONSE

**Reply by mdmintz** [reply]
Perhaps the issue is with the proxy itself? Otherwise it may be https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/794

**Reply by GTWHTX** [reply]
All proxies are good, wire is not being used. What else could be the problem?

**Reply by mdmintz** [reply]
Actually, I think I see the issue. I may have to make a code change and push a new release for this...

**Reply by GTWHTX** [reply]
Thank you, looking forward to it!

**Reply by mdmintz** [reply]
The current proxy-with-auth implementation uses this: https://stackoverflow.com/a/35293284/7058266

I'm proposing this change:

<img width="537" alt="Screenshot 2023-12-05 at 6 52 21 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/7c43cfc9-572f-4048-9081-4b11992a0cf8"><br />

(For the following proxy format: `proxy="user:pass@socks5://host:port"`)

I don't have an actual socks5 proxy-with-auth to test with, so I'm not sure if this will work.

Also, according to https://chromium.googlesource.com/chromium/src/+/HEAD/net/docs/proxy.md#SOCKSv5-proxy-scheme, if the above change isn't enough, I may need to change the `scheme` in that file to `SOCKSv5` (from `http`).

This is very speculative. Without a working socks5 proxy-with-auth to test with, I'm not sure if I'm on the right track. If you can determine if my change is enough, I can ship a new release for that quickly.

**Reply by GTWHTX** [reply]
You're right. That wasn't enough, this time i got ERR_NO_SUPPORTED_PROXIES. If you want, i can provide a list of socks5 proxies for you to test.

**Reply by mdmintz** [reply]
Thank you. With at least one known working socks5 proxy-with-auth to test with, I can keep playing around with that configuration file until it actually works. I couldn't find anything specific to socks5 proxy-with-auth on Stack Overflow. Only regular http/https proxy-with-auth.

**Reply by mdmintz** [reply]
Got this working:

```python
from seleniumbase import Driver

def set_proxy():
    driver.proxy = {
        "http": "socks5://7740735-all-country-IN-session-1:unfow9oi6@194.88.105.177:55328",
        "https": "socks5://7740735-all-country-IN-session-1:unfow9oi6@194.88.105.177:55328",
        "no_proxy": "localhost,127.0.0.1",
    }

driver = Driver(wire=True)
try:
    set_proxy()
    driver.get("https://ipinfo.io/")
    driver.sleep(5)
finally:
    driver.quit()
```

(And no code changes were needed for this!)

**Reply by GTWHTX** [reply]
What about UC mode without wire?

**Reply by mdmintz** [reply]
After 30 minutes of experimentation with https://stackoverflow.com/a/35293284/7058266, I'm still unable to get that algorithm working with socks5. That algorithm is used for setting a proxy without using selenium-wire. I may have to rethink the algorithm. I'm be doing a lot more experimentation with it. I feel like it could be a small change to make it work. Regular proxy-with-auth works with it. Not yet sure why socks5 proxy would be that much different. If you figure it out before I do, let me know so that I can update SeleniumBase with those changes.

**Reply by mdmintz** [reply]
I suppose one way would be figuring out how to fix the `selenium-wire` bugs that prevent combining that with UC Mode. Then UC mode could be activated, while setting the proxy with `selenium-wire`. So far, that has only led to detection if it works at all.
